test/selenium/sessions_ui_test.js

```diff
@@ -47,115 +54,69 @@ describe('sessions ui', function() {
     });
 
     describe('signup', function() {
-      var elements;
-      var screens;
 
-      beforeEach(function() {
-        screens = {
-          signup: driver.findElement(webdriver.By.id('signup')),
-          signupSuccess: driver.findElement(webdriver.By.id('signup-success'))
-        };
-
-        elements = {
-          pwd1: driver.findElement(webdriver.By.id('signup-pwd1')),
-          pwd2: driver.findElement(webdriver.By.id('signup-pwd2')),
-          set: driver.findElement(webdriver.By.id('signup-button'))
-        };
-      });
+        beforeEach(function() {
+            setUpWebapp = new SetUpWebapp(driver);
+            setUpPage = setUpWebapp.getSetUpView();
+        });
 
       it('should show the signup screen', function() {
-        return driver.wait(webdriver.until.elementIsVisible(screens.signup),
-                           3000);
-      });
-
-      it('should not show the signupSuccess screen', function(done) {
-        screens.signupSuccess.isDisplayed().then(function(visible) {
-          assert.equal(visible, false);
-          done();
+          return setUpPage.isSetUpView();
         });
-      });
 
-      it('should have the right fields', function () {
-        var types = {
-          pwd1: 'password',
-          pwd2: 'password',
-          set: 'submit'
-        };
-        var promises = Object.keys(elements).map(function(key) {
-          return elements[key].getAttribute('type').then(function(value) {
-            assert.equal(value, types[key]);
-          });
-        });
-        return Promise.all(promises);
+      it('should have the rights fields', function() {
+          return setUpPage.confirmAllFieldsArePresent();
       });
 
-      it('should reject non-matching passwords', function () {
-        return elements.pwd1.sendKeys('asdfasdf').then(function() {
-          return elements.pwd2.sendKeys('qwerqwer');
-        }).then(function() {
-          return elements.set.click();
-        }).then(function() {
-          return driver.wait(webdriver.until.alertIsPresent(), 5000);
-        }).then(function() {
-          return driver.switchTo().alert();
-        }).then(function(alert) {
-          return alert.getText().then(function(text) {
-            assert.equal(text, 'Passwords don\'t match! Please try again.');
-          }).then(function() {
-            alert.dismiss();
+      it('should reject non-matching passwords', function() {
+          
+          return setUpPage.typePassword('12345678').then(function(){
```

Comment:
*Proposed improvement for a follow up*

How about factorizing the failure of a login in 1 function in the View ? That would help in having function that reflect the user's intent in Views. For example:
```js
it('should reject non-matching passwords', function() {
  return setUpPage.signUpFailure(12345678, 123456).then(alertMessage) {
    assert.equal(alertMessage, errorPasswordDoNotMatch);
  });
})
```
```js
SetUpView.prototype.signUpFailure = function(password, confirmPassword) {
  this.accessors.passwordField.sendKeys(password).then(function() {
    this.accessors.confirmPasswordField.sendKeys(confirmPassword);
  }).then(function () {
 // and so on
}
```